ls1intum/Artemis
# `Programming exercises`: Fix internal server error when opening instructor code editor

Due to some Hibernate query issues, internal server errors (null pointer exception) can occur. I refactored the corresponding code using the same approach as in #3989

This should prevent the null pointer exception

### Checklist
#### General


#### Server
#### Changes affecting Programming Exercises


### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Programming Exercise

1. Navigate into the instructor page `Edit in Editor`
2. Open template, solution and tests
3. Make sure no exceptions occur

Important: should also be tested on `staging` because the error might not happen on single node environments

### Review Progress





#### Code Review
#### Manual Tests
